### PR TITLE
op-e2e: Add test to demonstrate duplicate claim bug

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

New failing e2e test that demonstrates a dispute game weakness because claims are considered duplicate if they have the same position and value even if they have different parents.

The test creates a dispute game that is attempting to delete a valid output root (ie the root claim is invalid) and we expect to get a `challenger_wins` result.  It lets the honest agent attack the root node as it should, then posts _two_ different invalid counter claims to the honest agent's attack. The first one is attacked by the honest agent creating a claim at gindex 8 with the correct trace value.

The honest response to the second invalid claim is _also_ to attack and create a claim at gindex 8 with the exact same correct trace value.  But it can't create that because it's a duplicate claim.  The game is then played out between the honest actor and dishonest but correct actor to push the attacked branch all the way down to the leaf claim which is countered.  In the end, everything is countered _except_ that second incorrect claim we created.  The defender will then win the game and the valid output root would be deleted.

Note that claim 7 is not countered:
```
        	            	7 - Parent: 1, Position: 4, Depth: 2, IndexAtDepth: 0 Trace Index: 268435455, Value: 0xcc00000000000000000000000000000000000000000000000000000000000000, Countered: false
```

Full game state:

```
Error Trace:	/Users/aj/Documents/code/optimism/op-e2e/e2eutils/disputegame/game_helper.go:113
        	            				/Users/aj/Documents/code/optimism/op-e2e/e2eutils/disputegame/game_helper.go:164
        	            				/Users/aj/Documents/code/optimism/op-e2e/faultproof_test.go:409
        	Error:      	Received unexpected error:
        	            	retrieve number of claims: dial tcp 127.0.0.1:58921: i/o timeout
        	Test:       	TestCannonDeleteValidOutputRoot
        	Messages:   	Did not find all claims countered
        	            	Game 0xA1bf5BB60A71f2a56a3056013c45fE9BcbfA3d76 (In Progress):
        	            	Claim count: 61
        	            	0 - Parent: 4294967295, Position: 1, Depth: 0, IndexAtDepth: 0 Trace Index: 1073741823, Value: 0x0199000000000000000000000000000000000000000000000000000000000000, Countered: true
        	            	1 - Parent: 0, Position: 2, Depth: 1, IndexAtDepth: 0 Trace Index: 536870911, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	2 - Parent: 1, Position: 6, Depth: 2, IndexAtDepth: 2 Trace Index: 805306367, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	3 - Parent: 2, Position: 14, Depth: 3, IndexAtDepth: 6 Trace Index: 939524095, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	4 - Parent: 3, Position: 30, Depth: 4, IndexAtDepth: 14 Trace Index: 1006632959, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	5 - Parent: 1, Position: 4, Depth: 2, IndexAtDepth: 0 Trace Index: 268435455, Value: 0xbb00000000000000000000000000000000000000000000000000000000000000, Countered: true
        	            	6 - Parent: 5, Position: 8, Depth: 3, IndexAtDepth: 0 Trace Index: 134217727, Value: 0x03db8405ab8070e7435645c365dc9d8759f2c3dca8c0de4a4770d4253c87da5c, Countered: true
        	            	7 - Parent: 1, Position: 4, Depth: 2, IndexAtDepth: 0 Trace Index: 268435455, Value: 0xcc00000000000000000000000000000000000000000000000000000000000000, Countered: false
        	            	8 - Parent: 4, Position: 62, Depth: 5, IndexAtDepth: 30 Trace Index: 1040187391, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	9 - Parent: 6, Position: 18, Depth: 4, IndexAtDepth: 2 Trace Index: 201326591, Value: 0x03e3a66e4dd58fbe59a3208b0505cbc7d33c94cb15d0620289599cf4a73a393a, Countered: true
        	            	10 - Parent: 8, Position: 126, Depth: 6, IndexAtDepth: 62 Trace Index: 1056964607, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	11 - Parent: 9, Position: 38, Depth: 5, IndexAtDepth: 6 Trace Index: 234881023, Value: 0x0330ea0cc76586dd9e506897c3751eb14388e93aee7449f63dfffcf81a70b995, Countered: true
        	            	12 - Parent: 10, Position: 254, Depth: 7, IndexAtDepth: 126 Trace Index: 1065353215, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	13 - Parent: 11, Position: 78, Depth: 6, IndexAtDepth: 14 Trace Index: 251658239, Value: 0x03f5bb5393a0f985b97e9341400248484581de3083449ae4ef3a6f8c5758b090, Countered: true
        	            	14 - Parent: 12, Position: 510, Depth: 8, IndexAtDepth: 254 Trace Index: 1069547519, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	15 - Parent: 13, Position: 158, Depth: 7, IndexAtDepth: 30 Trace Index: 260046847, Value: 0x0349d8ce37efad71ccfd1dc771406a8e1cf9cdc3caebd5286eaaf6c8d1ccd039, Countered: true
        	            	16 - Parent: 14, Position: 1022, Depth: 9, IndexAtDepth: 510 Trace Index: 1071644671, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	17 - Parent: 15, Position: 318, Depth: 8, IndexAtDepth: 62 Trace Index: 264241151, Value: 0x033e47319c824a5daceea02a07af812dfe9889a10766f3d1abb0960537715154, Countered: true
        	            	18 - Parent: 16, Position: 2046, Depth: 10, IndexAtDepth: 1022 Trace Index: 1072693247, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	19 - Parent: 17, Position: 638, Depth: 9, IndexAtDepth: 126 Trace Index: 266338303, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	20 - Parent: 18, Position: 4094, Depth: 11, IndexAtDepth: 2046 Trace Index: 1073217535, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	21 - Parent: 19, Position: 1278, Depth: 10, IndexAtDepth: 254 Trace Index: 267386879, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	22 - Parent: 20, Position: 8190, Depth: 12, IndexAtDepth: 4094 Trace Index: 1073479679, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	23 - Parent: 21, Position: 2558, Depth: 11, IndexAtDepth: 510 Trace Index: 267911167, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	24 - Parent: 22, Position: 16382, Depth: 13, IndexAtDepth: 8190 Trace Index: 1073610751, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	25 - Parent: 23, Position: 5118, Depth: 12, IndexAtDepth: 1022 Trace Index: 268173311, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	26 - Parent: 24, Position: 32766, Depth: 14, IndexAtDepth: 16382 Trace Index: 1073676287, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	27 - Parent: 25, Position: 10238, Depth: 13, IndexAtDepth: 2046 Trace Index: 268304383, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	28 - Parent: 26, Position: 65534, Depth: 15, IndexAtDepth: 32766 Trace Index: 1073709055, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	29 - Parent: 27, Position: 20478, Depth: 14, IndexAtDepth: 4094 Trace Index: 268369919, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	30 - Parent: 28, Position: 131070, Depth: 16, IndexAtDepth: 65534 Trace Index: 1073725439, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	31 - Parent: 29, Position: 40958, Depth: 15, IndexAtDepth: 8190 Trace Index: 268402687, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	32 - Parent: 31, Position: 81918, Depth: 16, IndexAtDepth: 16382 Trace Index: 268419071, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	33 - Parent: 30, Position: 262142, Depth: 17, IndexAtDepth: 131070 Trace Index: 1073733631, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	34 - Parent: 33, Position: 524286, Depth: 18, IndexAtDepth: 262142 Trace Index: 1073737727, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	35 - Parent: 32, Position: 163838, Depth: 17, IndexAtDepth: 32766 Trace Index: 268427263, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	36 - Parent: 35, Position: 327678, Depth: 18, IndexAtDepth: 65534 Trace Index: 268431359, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	37 - Parent: 34, Position: 1048574, Depth: 19, IndexAtDepth: 524286 Trace Index: 1073739775, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	38 - Parent: 37, Position: 2097150, Depth: 20, IndexAtDepth: 1048574 Trace Index: 1073740799, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	39 - Parent: 36, Position: 655358, Depth: 19, IndexAtDepth: 131070 Trace Index: 268433407, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	40 - Parent: 39, Position: 1310718, Depth: 20, IndexAtDepth: 262142 Trace Index: 268434431, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	41 - Parent: 38, Position: 4194302, Depth: 21, IndexAtDepth: 2097150 Trace Index: 1073741311, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	42 - Parent: 40, Position: 2621438, Depth: 21, IndexAtDepth: 524286 Trace Index: 268434943, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	43 - Parent: 41, Position: 8388606, Depth: 22, IndexAtDepth: 4194302 Trace Index: 1073741567, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	44 - Parent: 42, Position: 5242878, Depth: 22, IndexAtDepth: 1048574 Trace Index: 268435199, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	45 - Parent: 43, Position: 16777214, Depth: 23, IndexAtDepth: 8388606 Trace Index: 1073741695, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	46 - Parent: 45, Position: 33554430, Depth: 24, IndexAtDepth: 16777214 Trace Index: 1073741759, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	47 - Parent: 44, Position: 10485758, Depth: 23, IndexAtDepth: 2097150 Trace Index: 268435327, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	48 - Parent: 47, Position: 20971518, Depth: 24, IndexAtDepth: 4194302 Trace Index: 268435391, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	49 - Parent: 46, Position: 67108862, Depth: 25, IndexAtDepth: 33554430 Trace Index: 1073741791, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	50 - Parent: 49, Position: 134217726, Depth: 26, IndexAtDepth: 67108862 Trace Index: 1073741807, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	51 - Parent: 48, Position: 41943038, Depth: 25, IndexAtDepth: 8388606 Trace Index: 268435423, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	52 - Parent: 51, Position: 83886078, Depth: 26, IndexAtDepth: 16777214 Trace Index: 268435439, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	53 - Parent: 50, Position: 268435454, Depth: 27, IndexAtDepth: 134217726 Trace Index: 1073741815, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	54 - Parent: 53, Position: 536870910, Depth: 28, IndexAtDepth: 268435454 Trace Index: 1073741819, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	55 - Parent: 52, Position: 167772158, Depth: 27, IndexAtDepth: 33554430 Trace Index: 268435447, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	56 - Parent: 55, Position: 335544318, Depth: 28, IndexAtDepth: 67108862 Trace Index: 268435451, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	57 - Parent: 54, Position: 1073741822, Depth: 29, IndexAtDepth: 536870910 Trace Index: 1073741821, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	58 - Parent: 57, Position: 2147483646, Depth: 30, IndexAtDepth: 1073741822 Trace Index: 1073741822, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	59 - Parent: 56, Position: 671088638, Depth: 29, IndexAtDepth: 134217726 Trace Index: 268435453, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
        	            	60 - Parent: 59, Position: 1342177278, Depth: 30, IndexAtDepth: 268435454 Trace Index: 268435454, Value: 0x00a49e44854f8dbda380da5c3a815b6ead9bfd1f7b2a53911c3ff8f72fa4b7b4, Countered: true
```

### CODE REVIEW

File: op-e2e/faultproof_test.go [hunk 1]
```diff
@@ -341,6 +341,81 @@ func TestCannonProposedOutputRootInvalid(t *testing.T) {
...
+
+	// Now that there is already a counter claim at gindex 8 using the correct hash, counter the honest actors
+	// first claim (depth 1, gindex 2) again using an invalid hash.
+	game.Attack(ctx, 1, common.Hash{0xbb})
+
+	// The honest challenger should counter that move by attacking with the correct value, but that will mean
+	// creating a second claim at gindex 8 with the same correct value.
+	game.WaitForClaim(ctx, 1, common.Hash{0xbb}, true)
+
+	// Then post a second attack on the root claim that is completely invalid.
```
Comment (Inphi) [inline op-e2e/faultproof_test.go:388]:
```suggestion
	// Then post a second attack on the honest actors first claim using an invalid hash
```

File: op-e2e/faultproof_test.go [hunk 2]
```diff
@@ -341,6 +341,81 @@ func TestCannonProposedOutputRootInvalid(t *testing.T) {
 	}
 }
 
+func TestCannonDeleteValidOutputRoot(t *testing.T) {
+	InitParallel(t)
```
Comment (Inphi) [inline op-e2e/faultproof_test.go:345]: missing test skip

**Review by Inphi — COMMENTED** [review]
Yup, this illustrates the issue well. Thanks for adding it in.
Needs a `t.Skip` before we can merge as the defender wins this scenario.

**Comment by ajsutton** [comment]
Closing as this is now covered by the exhaustive response test.

**Pull request closed without merging**